keps/sig-scheduling/2044-topology-awareness-in-kube-scheduler/README.md

```diff
@@ -0,0 +1,767 @@
+worker node takes that into account.
+
+## Design Details
+
+- add a new flag in Kubelet called `ExposeNodeResourceTopology` in the kubelet config or command line argument called `expose-noderesourcetopology` which allows 
+  the user to specify when they would like Kubelet to compute and expose resource hardware topology information.
+- The `ExposeNodeResourceTopology` flag is received from the kubelet config/command line args is propogated to the Container Manager.
+- Based on the resources allocated to a pod, the topology associated with the resources is evaluated and populated as part of the NodeResourceTopology.
+- Kubelet will collect information about resources allocated to running pods along with their topology, based on allocatable resources of the node and consumed
+  resources by pods it will populate available resources with the associated topology information in NodeResourceTopology, where a NodeResourceTopology instance
+  would represent a worker node.
+  The name of the CRD instance is the name of the worker node
+- A new in-tree scheduler plugin `NodeResourceTopologyMatch` is created that makes topology-aware placement decisions implements a simplified version of Topology
+  Manager as part of the filter extension point to filter out nodes that are not suitable for the workload based on the resource request and the obtained 
+  NodeResourceTopology information corresponding to that worker node. The scoring extension point to determine a score based on a configurable strategy
+  would be done as a Beta feature. 
+- The scheduler plugin `NodeResourceTopologyMatch` would be disabled by default and when enabled would also check for the ability to run pod only
+  in case of Topology Manager policy of single-numa-node policy is configured on the node. Since it is the most strict policy, it implies that the launch
+  on the node with other existing policies will be successful if the condition for single-numa-node policy passed for the worker node.
+ 
+
+### Design Consideration
+
+By default Kubelet would be resposible to fill the NodeResourceTopology instances, but a design consideration is to architect the
+solution in such a way to allow future extension to facilitate the cloud administratore to disable the feature and introduce a custom external agents to fill the NodeResourceTopology data.
+
+### Changes to the API
+
+Code responsible for working with NodeResourceTopology API will be placed in the staging directory
+at path staging/src/k8s.io/api/node/v1/types.go.
+
+```go
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeResourceTopology is a specification for a hardware resources
+type NodeResourceTopology struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	TopologyPolicies []string 		   `json:"topologyPolicies"`
```

Comment:
why is this a slice?

Reply:
Can you list the policies? And explain them?
like：SingleNUMANodeContainerLevel、SingleNUMANodePodLevel ？